Makefile

```diff
@@ -308,15 +298,26 @@ test-unit: clean-cov generate fmt vet ## Run Unit tests.
 
 ##@ Build
 
+WASM_SHIM = $(PROJECT_PATH)/kuadrant-wasm-shim
+WASM_SHIM_VERSION ?= v0.4.0-alpha.3
+$(WASM_SHIM):
+	@{ \
+	ASSET_ID=$$(curl -s -H "Accept: application/vnd.github.v3.raw" \
+		https://api.github.com/repos/Kuadrant/wasm-shim/releases | \
+		jq ". | map(select(.tag_name == \"$(WASM_SHIM_VERSION)\"))[0].assets | map(select(.name == \"kuadrant-wasm-shim-$(WASM_SHIM_VERSION)\"))[0].id" \
+	) \
+	&& echo "Downloading kuadrant-wasm-shim@$(WASM_SHIM_VERSION) from https://api.github.com/repos/Kuadrant/wasm-shim/releases/assets/$${ASSET_ID}" \
+	&& curl -sSLo $@ -H "Accept: application/octet-stream" https://api.github.com/repos/Kuadrant/wasm-shim/releases/assets/$${ASSET_ID}; \
+	}
+	sha256sum $@
```

Comment:
I wonder if `local-cleanup` should also remove the `kuadrant-wasm-shim` binary, otherwise it won't ever get updated.

```sh
❯ make wasm-shim
Downloading kuadrant-wasm-shim@v0.4.0-alpha.3 from https://api.github.com/repos/Kuadrant/wasm-shim/releases/assets/173311367

sha256sum ./kuadrant-operator/kuadrant-wasm-shim
12879567faee3d2625a4998f6a4e622ded01163133bdef2b539bfa62f921cdd8  ./kuadrant-wasm-shim

❯ make wasm-shim WASM_SHIM_VERSION=v0.4.0-alpha.4
make: Nothing to be done for `wasm-shim'.
```

Reply:
done